pkg/kubectl/cmd/cmd_printing_test.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+func TestIllegalPackageSourceCheckerThroughPrintFlags(t *testing.T) {
+	testCases := []struct {
+		name                 string
+		expectInternalObjErr bool
+		output               string
+		obj                  runtime.Object
+		expectedOutput       string
+	}{
+		{
+			name:                 "success printer: object containing package path beginning with forbidden prefix is rejected",
+			expectInternalObjErr: true,
+			obj:                  internalPod(),
+		},
+		{
+			name:                 "success printer: object containing package path with no forbidden prefix returns no error",
+			expectInternalObjErr: false,
+			obj:                  externalPod(),
+			output:               "",
+			expectedOutput:       "pod/foo succeeded\n",
+		},
+		{
+			name:                 "name printer: object containing package path beginning with forbidden prefix is rejected",
+			expectInternalObjErr: true,
+			output:               "name",
+			obj:                  internalPod(),
+		},
+		{
+			name:                 "json printer: json printer is wrapped in a versioned printer - internal obj should be converted with no error",
```

Comment:
@deads2k the json / yaml test cases here do not fail when an internal object is passed - as retrieving these printers through `PrintFlags#ToPrinter` returns the printer wrapped in a VersionedPrinter

Reply:
> @deads2k the json / yaml test cases here do not fail when an internal object is passed - as retrieving these printers through PrintFlags#ToPrinter returns the printer wrapped in a VersionedPrinter

Add a printer to each test case and make sure yaml, json, and template printers all fail on internal inputs.

Reply:
@deads2k 
> Add a printer to each test case and make sure yaml, json, and template printers all fail on internal inputs.

This is done in the test below this one